contracts/src/emergency/sequencer_uptime_feed.cairo

```diff
@@ -0,0 +1,358 @@
+            Result::Ok(
+                RoundTransmission {
+                    answer: storage_read_syscall(
+                        address_domain, storage_address_from_base_and_offset(base, 0_u8)
+                    )?.try_into().unwrap(),
+                    block_num: storage_read_syscall(
+                        address_domain, storage_address_from_base_and_offset(base, 1_u8)
+                    )?.try_into().unwrap(),
+                    started_at: storage_read_syscall(
+                        address_domain, storage_address_from_base_and_offset(base, 2_u8)
+                    )?.try_into().unwrap(),
+                    updated_at: storage_read_syscall(
+                        address_domain, storage_address_from_base_and_offset(base, 3_u8)
+                    )?.try_into().unwrap(),
+                }
+            )
+        }
+
+        fn write(
+            address_domain: u32, base: StorageBaseAddress, value: RoundTransmission
+        ) -> SyscallResult::<()> {
+            storage_write_syscall(
+                address_domain,
+                storage_address_from_base_and_offset(base, 0_u8),
+                value.answer.into()
+            )?;
+            storage_write_syscall(
+                address_domain,
+                storage_address_from_base_and_offset(base, 1_u8),
+                value.block_num.into()
+            )?;
+            storage_write_syscall(
+                address_domain,
+                storage_address_from_base_and_offset(base, 2_u8),
+                value.started_at.into()
+            )?;
+            storage_write_syscall(
+                address_domain,
+                storage_address_from_base_and_offset(base, 3_u8),
+                value.updated_at.into()
+            )
+        }
+    }
+
+
+    struct Storage {
+        // l1 sender is an ethereum address
+        _l1_sender: u256,
+        // maps round id to round transmission
+        _round_transmissions: LegacyMap<u128, RoundTransmission>,
+        _latest_round_id: u128,
+    }
+
+    #[event]
+    fn RoundUpdated(status: bool, updated_at: u64) {}
+
+    #[event]
+    fn NewRound(round_id: u128, started_by: ContractAddress, started_at: u64) {}
+
+    #[event]
+    fn AnswerUpdated(current: bool, round_id: u128, timestamp: u64) {}
+
+    #[event]
+    fn UpdateIgnored(latest_status: bool, latest_timestamp: u64, incoming_status: bool, incoming_timestamp: u64) {}
+
+    #[event]
+    fn L1SenderTransferred(from_address: u256, to_address: u256) {}
+
+    impl Aggregator of IAggregator {
+        fn latest_round_data() -> Round {
+            require_access();
+            let latest_round_id = _latest_round_id::read();
+            let round_transmission = _round_transmissions::read(latest_round_id);
+            // bool -> felt252 -> u128
+            let answer: u128 = round_transmission.answer.into().try_into().unwrap();
+            Round {
```

Comment:
Round is the external struct returned by the Proxy, so we have to do some type conversions for round_id and answer. This is because internally round_id uses u128 and answer uses bool whereas externally the AggregatorProxy defines round_id as felt252 and answer as u128.